tools/ci/java-ci-tools/src/main/java/org/apache/flink/tools/ci/suffixcheck/ScalaSuffixChecker.java

```diff
@@ -96,40 +98,35 @@ private static ParseResult parseMavenOutput(final Path path) throws IOException
         final Set<String> cleanModules = new HashSet<>();
         final Set<String> infectedModules = new HashSet<>();
 
-        try (final BufferedReader bufferedReader =
-                Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
+        final Map<String, Set<Dependency>> dependenciesByModule =
+                DependencyParser.parseDependencyTree(path);
 
-            String line;
-            while ((line = bufferedReader.readLine()) != null) {
-                final Matcher matcher = moduleNamePattern.matcher(line);
-                if (matcher.matches()) {
-                    final String moduleName = stripScalaSuffix(matcher.group(1));
+        for (String module : dependenciesByModule.keySet()) {
+            final String moduleName = stripScalaSuffix(module);
+            {
+                {
```

Comment:
These are only here to make the review easier (keeping the indentation levels the same), and are removed in the second commit.